tests/unit/config/test_configcommands.py

```diff
@@ -282,6 +282,54 @@ def test_set_toggle_print(self, commands, config_stub, message_mock):
         assert msg.text == 'auto_save.session = true'
 
 
+class TestAdd:
+
+    """Test :config-add-list and :config-add-dict."""
+
+    @pytest.mark.parametrize('name', ['content.host_blocking.whitelist',
+                                      'history_gap_interval'])
+    @pytest.mark.parametrize('temp', [True, False])
+    @pytest.mark.parametrize('value', ['test1', 'test2', '', None])
+    def test_add_list(self, commands, config_stub, yaml_value, name, temp, value):
+        opt_type = config_stub.get_opt(name).typ
+
+        try:
+            commands.config_add_list(name, value, temp=temp)
+        except cmdexc.CommandError:
+            # We attempted to add to the dictionary with replace as false.
+            valid_list_types = (configtypes.List, configtypes.ListOrValue)
+            assert not isinstance(opt_type, valid_list_types) or not value
+            return
+
+        assert str(config_stub.get(name)[-1]) == value
+        if temp:
+            assert yaml_value(name) == configutils.UNSET
+        else:
+            assert yaml_value(name)[-1] == value
+
+    @pytest.mark.parametrize('name', ['aliases', 'history_gap_interval'])
+    @pytest.mark.parametrize('key', ['w', 'missingkey'])
+    @pytest.mark.parametrize('value', ['test1', 'test2'])
+    @pytest.mark.parametrize('temp', [True, False])
+    @pytest.mark.parametrize('replace', [True, False])
+    def test_add_dict(self, commands, config_stub, yaml_value, name, key,
+                      value, temp, replace):
+        opt_type = config_stub.get_opt(name).typ
+
+        try:
+            commands.config_add_dict(name, key, value, temp=temp, replace=replace)
+        except cmdexc.CommandError:
```

Comment:
Those should be separate tests, which use `pytest.raises` (and maybe even check for the correct error message).